src/constants/groups.ts

```diff
@@ -1,6 +1,12 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
-export default {
+
+import {SyncableType} from 'types/groups';
+
+// See LICENSE.txt for license information.
+const Groups: {[key: string]: SyncableType} = {
```

Comment:
This is actually a type change since the export is going from the type `{SYNCABLE_TYPE_TEAM: X, SYNCABLE_TYPE_CHANNEL: X}` to `{[key: string]: X}`, so `Groups.Foobarasdf` goes from being a type error to just returning undefined.

In this case, I think we need to cast the `'team'` and `'channel'` literals to `SyncableType` or find some way to tell TypeScript that it should do that TypeScript without using a type like this that makes keys potentially be undefined

Reply:
Also, I think this is the big thing you're wanting to fix here since the problem seems to be that Groups.SYNCABLE_TYPE_TEAM is not actually a SyncableType currently

Reply:
Suggestion - replace all {[key: string]: SyncableType} / Dictionary / OneToOneRelashionship stuff that we already have with the built in 'Record' type.

Reply:
@reflog That's a much bigger undertaking, and I'd like to do it. But without this change we are currently blocking this contributor PR: https://github.com/mattermost/mattermost-webapp/pull/6924.

Can we merge as-is and come back to fixing this properly later?

Reply:
Agree on doing that separately. We know that there's issues with the types defined here, but we haven't been able to devote much time to fixing them as a whole